insights/combiners/httpd_conf.py

```diff
@@ -114,42 +117,64 @@ def __init__(self, local, shared):
-                                                                file_path)
+                        values = [self.ParsedData(a.value, a.line, file_name, file_path)
+                                  for a in pd]
+                        self.data[section][k] = values
                 else:
                     # For directive
-                    self.data[option] = self.ParsedData(parsed_data.value, parsed_data.line,
-                                                        file_name, file_path)
+                    values = [self.ParsedData(a.value, a.line, file_name, file_path)
+                              for a in parsed_data]
+                    self.data[option] = values
 
-    def get_valid_setting(self, directive, section=None):
+    def get_setting_list(self, directive, section=None):
         """
-        Return the valid value of specified directive.
+        Returns the parsed data of the specified directive as a list of named tuples.
 
         Parameters:
             directive (str): The directive to look for.
             section (str): The section if the directive belongs to one.
 
         Returns:
-            (tuple): ('the value of the directive', 'the configuration file') if directive exists,
-                     else None
+            (list): List of named tuples ParsedData, in order how they are parsed. If directive
+                    does not exist, it returns None.
         """
         if section:
-            parsed_data = self.data.get(section, {}).get(directive)
-            return (parsed_data.value, parsed_data.file_name) if parsed_data is not None else None
-        parsed_data = self.data.get(directive)
-        return (parsed_data.value, parsed_data.file_name) if parsed_data is not None else None
+            return self.data.get(section, {}).get(directive)
+        return self.data.get(directive)
 
-    def get_valid_setting_full(self, directive, section=None):
+    def get_active_setting(self, directive, section=None):
         """
-        Return the valid parsed data of specified directive as a named tuple with .
+        Return the active parsed setting of the specified directive as a named tuple. It is the
+        last parsed value and for most directives it is also the active setting.
 
         Parameters:
             directive (str): The directive to look for.
             section (str): The section if the directive belongs to one.
 
         Returns:
-            (namedtuple): Named tuple ParsedData.
+            (namedtuple): Named tuple ParsedData if directive exists, else None.
         """
-        if section:
-            return self.data.get(section, {}).get(directive)
-        return self.data.get(directive)
+        values_list = self.get_setting_list(directive, section)
+        if values_list is None:
+            return None
+        else:
+            return values_list[-1]
+
+    def get_valid_setting(self, directive, section=None):
```

Comment:
The explanation makes sense.
I didn't find a comment about get_valid_setting() being a legacy method.
While get_valid_setting and get_active_setting are so similar, there's no explanation whether there is any human meaning difference between "active" and "valid" and the only way for me to find out is to spot the different representation of the output data.
I suggest adding a short comment about it being a legacy method.